servers/rendering/renderer_rd/storage_rd/material_storage.cpp

```diff
@@ -2411,15 +2414,15 @@ MaterialStorage::Samplers MaterialStorage::samplers_rd_allocate(float p_mipmap_b
 					sampler_state.min_filter = RD::SAMPLER_FILTER_NEAREST;
 					sampler_state.mip_filter = mip_filter;
 					sampler_state.lod_bias = samplers.mipmap_bias;
-					sampler_state.use_anisotropy = true;
+					sampler_state.use_anisotropy = !Math::is_equal_approx(anisotropy_max, 1.0f) && !is_intel_gpu;
```

Comment:
This is needlessly obfuscated. You can just check `if (anisotropic_filtering_level == VIEWPORT_ANISOTROPY_DISABLED)`. Its a less expensive check and it is absolutely clear what you are checking